File: cusPlugins/PaymentPlugin/www/PaymentPlugin.js
var exec = require('cordova/exec');
const pluginName = 'PaymentPlugin'

exports.success = function () {
    return "1";
};

exports.failed = function () {
    return "-1";
};

exports.cancel = function () {
    return "0";
};

exports.checkIsReadyToPay  = function (resultCallback) {
    exec(resultCallback, function() {}, pluginName, 'checkIsReadyToPay', []);
};

exports.requestPayment  = function (price) {
    exec(function() {}, function() {}, pluginName, 'requestPayment', [price]);
};

File: plugins/webcomm-cordova-plugin-payment/src/ios/PaymentPlugin.swift
import JavaScriptCore

let handler = PaymentHandler()

@objc(PaymentPlugin)
class PaymentPlugin : CDVPlugin {
    
    
    @objc(checkIsReadyToPay:)
    func checkIsReadyToPay(_ command: CDVInvokedUrlCommand) {
        let result = PaymentHandler.applePayStatus()
        let pluginResult = CDVPluginResult(status: CDVCommandStatus_OK, messageAs: result.canSetupCards && result.canSetupCards ? 1: 0)
        self.commandDelegate!.send(pluginResult, callbackId: command.callbackId)
    }
    
    @objc(requestPayment:)
    func requestPayment(_ command: CDVInvokedUrlCommand) {
        let money = command.arguments ?? [0, 0]
        handler.startPayment(amount: money) {
            (success) in
            //            let pluginResult = CDVPluginResult(status: CDVCommandStatus_OK, messageAs: success ? 1: 0)
            //            self.commandDelegate!.send(pluginResult, callbackId: command.callbackId)
            self.commandDelegate!.evalJs("eventsHandler.exec(\(success ? "1": "0"));")
        }
    }
}
